diff --git a/dispatch/dispatch.h b/dispatch/dispatch.h
index 9869bc2..89412ab 100644
--- a/dispatch/dispatch.h
+++ b/dispatch/dispatch.h
@@ -45,7 +45,7 @@
 
 #ifdef __linux__
 #if __has_feature(modules)
-#include <stdio.h> // for off_t (to match Glibc.modulemap)
+// #include <stdio.h> // for off_t (to match Glibc.modulemap)
 #endif
 #define DISPATCH_LINUX_UNAVAILABLE() \
 		__DISPATCH_UNAVAILABLE("This interface is unavailable on linux systems")
diff --git a/src/shims/linux_stubs.h b/src/shims/linux_stubs.h
index e133049..b3465ed 100644
--- a/src/shims/linux_stubs.h
+++ b/src/shims/linux_stubs.h
@@ -98,4 +98,8 @@ typedef void (*dispatch_mach_msg_destructor_t)(void*);
 
 #define NOTE_VM_PRESSURE ({LINUX_PORT_ERROR(); 0;})
 
+#ifndef PAGE_SIZE
+#define PAGE_SIZE 4096
+#endif
+
 #endif
